core/src/main/java/com/datastax/oss/driver/internal/core/config/typesafe/DefaultDriverConfigLoader.java

```diff
@@ -127,7 +127,11 @@ public void close() {
    * <p>In the general case, use of this is not recommended, but it may be useful in situations
    * where configuration must be defined at runtime or is derived from some other configuration
    * source.
+   *
+   * @deprecated this feature is now available in the public API. Use {@link
+   *     DriverConfigLoader#programmaticBuilder()} instead.
    */
+  @Deprecated
```

Comment:
Can't we simply remove it, since it's an internal package?

Reply:
Even though we allow ourselves to break the internal API, I think we should avoid it as much as possible.

Reply:
It bothers me that we already have deprecated stuff in a 4.0.1 release. I know we recommended some people out there to start using this internal API, but honestly atm I would say that no more than 2-3 users are actively using it.